core/include/scipp/core/element/arithmetic.h

```diff
@@ -47,10 +66,21 @@ template <class... Ts> struct divide_types_t {
   using types = arithmetic_type_pairs;
 };
 
-constexpr auto plus =
-    overloaded{add_types_t{}, [](const auto a, const auto b) { return a + b; }};
-constexpr auto minus =
-    overloaded{add_types_t{}, [](const auto a, const auto b) { return a - b; }};
+constexpr auto plus = overloaded{
+    add_types_t{}, [](const auto a, const auto b) { return a + b; },
+    [](const scipp::core::time_point &a, const scipp::core::time_point &b) {
+      // a + b for time_point is undefined. Can't throw from here
+      // so return false
+      (void)a;
+      (void)b;
```

Comment:
rather than `void` why not just leave the variable out of the function definition? `[](const scipp::core::time_point &a, const scipp::core::time_point &b) {}`

Reply:
because then I get warnings for unused 'a' and 'b', which I wanted to avoid.

Reply:
Sorry copied function signature, but did not alter. Should have said. `[](const scipp::core::time_point &, const scipp::core::time_point &) {}`

Reply:
@OwenArnold We were discussing yesterday that we need our own `time_point`, so a bit of this here will be removed, since not necessary any more, so hold back with reviewing more.

Reply:
This is now moot, since we changed the `time_point` class.